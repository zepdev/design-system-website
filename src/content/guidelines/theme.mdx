---
label: Guidelines
title: Theme
---

# Theme

Themes are used to customize existing elements to fit the specific style of a strategic business unit or product. There is no base color for the Zeppelin-element-library, almost all elements require that a theme class is present. At the moment the Zeppelin Design System offers three main themes. Developers can customize all elements by adding the theme class to any html tag (`<main>`,`<body>`,`<div>`, ...) as long it wraps all elements you want to use.

```js
<main class="theme-zeppelin" id="app-root">
  ...
</main>
```

We currently offer three main themes;

```js
// Zeppelin Indigo Blue
<div class="theme-zeppelin" >...</div>

// CAT Yellow
<div class="theme-cat" >...</div>

// Rental Red
<div class="theme-rental" >...</div>
```
