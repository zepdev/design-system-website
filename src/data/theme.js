module.exports = {
  breakpoints: {
    xs: '36em' /* 576px */,
    s: '48em' /* 768px */,
    m: '62em' /* 992px */,
    l: '75em' /* 1200px */,
    xl: '105em' /* 1680px */,
    xxl: '120em' /* 1920px */,
  },
  typography: {
    typefaces: {
      roboto: "'Roboto', Arial, Helvetica, sans-serif",
      teko: "Teko', sans-serif",
    },
    basics: {
      fontSizeBase: {
        specification: 'font-size-base',
        px: '16px',
        rem: '1rem',
        weight: null,
      },
      lineHeightBase: {
        specification: 'line-height-base',
        px: '24px',
        rem: '1.5rem',
        weight: null,
      },
      fontWeightBase: {
        specification: 'font-weight-base',
        px: null,
        rem: null,
        weight: 400,
      },
    },
    headings: {
      normal: {
        'zep-typo--normal-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-5': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-6': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-7': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-8': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      display: {
        'zep-typo--display-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      special: {
        'zep-typo--special-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
    },
  },
  colors: {
    primary: {
      indigoBlue: {
        name: 'indigo-blue',
        hex: '#27166f',
        rgb: '39/22/111',
        hsb: null,
      },
      catYellow: {
        name: 'cat-yellow',
        hex: '#f0b823',
        rgb: '240/184/35',
        hsb: null,
      },
      rentalRed: {
        name: 'rental-red',
        hex: '#af0e0e',
        rgb: '175/14/14',
        hsb: null,
      },
    },
    gray: {
      grayDark: {
        name: 'gray-dark',
        hex: '#101111',
        rgb: '16/17/17',
        hsb: null,
      },
      grayMidDark: {
        name: 'gray-mid-dark',
        hex: '#373d3f',
        rgb: '55/61/63',
        hsb: null,
      },
      gray: { name: 'gray', hex: '#464a4c', rgb: '70/74/76', hsb: null },
      grayLight: {
        name: 'gray-light',
        hex: '#6f7c80',
        rgb: '111/124/128',
        hsb: null,
      },
      grayLighter: {
        name: 'gray-lighter',
        hex: '#eceeef',
        rgb: '236/238/239',
        hsb: null,
      },
      grayMid: {
        name: 'gray-mid',
        hex: '#f2f3f4',
        rgb: '242/243/244',
        hsb: null,
      },
      grayLightest: {
        name: 'gray-lightest',
        hex: '#f7f7f9',
        rgb: '247/247/249',
        hsb: null,
      },
      white: { name: 'white', hex: '#ffffff', rgb: '255/255/255', hsb: null },
    },
    support: {
      alert: { name: 'alert', hex: '#ff6562', rgb: '255/101/98', hsb: null },
      warning: {
        name: 'warning',
        hex: '#f0ad4e',
        rgb: '240,/73/78',
        hsb: null,
      },
      success: { name: 'success', hex: '#5cb85c', rgb: '92/184/92', hsb: null },
      info: { name: 'info', hex: '#5bc0de', rgb: '91/192/222', hsb: null },
    },
  },
  spacing: {
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 24,
    xl: 32,
    xxl: 48,
  },
}
