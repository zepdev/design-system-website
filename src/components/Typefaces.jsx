import React from 'react'
import PropTypes from 'prop-types'
import withStyles from 'react-jss'
import LaunchIcon from './icons/LaunchIcon'
import theme from '../data/theme'

const styles = theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
    marginBottom: theme.spacing.xl * 2,
  },
  container: {
    flexBasis: '48%',
  },
  titleContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    height: 136,
    background: theme.colors.gray.grayLighter.hex,
  },
  title: {
    fontSize: 28,
  },
  buttonContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    background: theme.colors.gray.grayLightest.hex,
    paddingTop: theme.spacing.m,
    paddingBottom: theme.spacing.m,
    paddingLeft: theme.spacing.l,
    paddingRight: theme.spacing.l,
    borderTop: `1px solid ${ theme.colors.gray.grayLight.hex }`,
  },
  text: {
    fontSize: 12,
    textTransform: 'capitalize',
  },
  link: {
    color: 'inherit',
    '&:hover': {
      color: theme.colors.primary.indigoBlue.hex,
    },
  },
})

const Typefaces = ({ classes }) => (
  <div className={classes.root}>
    {Object.keys(theme.typography.typefaces).map((elem, idx) => (
      <div className={classes.container} key={`typeface${ idx }`}>
        <div className={classes.titleContainer}>
          <p className={classes.title} style={{ fontFamily: elem }}>
            Lorem Ipsum
          </p>
        </div>
        <div className={classes.buttonContainer}>
          <p className={classes.text}>{elem} Font Family</p>
          <a
            href={`https://fonts.google.com/specimen/${ elem.charAt(0).toUpperCase() +
              elem.slice(1).toLowerCase() }`}
            target="_blank"
            rel="noopener"
            aria-label="Launch"
            className={classes.link}
          >
            <LaunchIcon width="24" height="24" />
          </a>
        </div>
      </div>
    ))}
  </div>
)

Typefaces.propTypes = {
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(Typefaces)
