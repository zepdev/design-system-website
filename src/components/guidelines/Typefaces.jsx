import React from 'react'
import PropTypes from 'prop-types'
import withStyles from 'react-jss'
import OpenInNewIcon from '../icons/OpenInNewIcon'
import theme from 'zeppelin-element-library/bundle/themes/theme.json'

const styles = theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-between',
    marginBottom: `${ theme.spacing.component.xxl.rem }rem`,
  },
  container: {
    flexBasis: '100%',
    paddingBottom: `${ theme.spacing.component.m.rem }rem`,
  },
  [`@media (min-width: ${ theme.breakpoints.s })`]: {
    container: {
      flexBasis: '48%',
      paddingBottom: 0,
    },
  },
  titleContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    height: 136,
    background: theme.colors.gray.grayLighter.hex,
  },
  title: {
    fontSize: 28,
  },
  buttonContainer: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between',
    background: theme.colors.gray.grayLightest.hex,
    paddingTop: theme.spacing.component.s.px,
    paddingBottom: theme.spacing.component.s.px,
    paddingLeft: theme.spacing.component.l.px,
    paddingRight: theme.spacing.component.l.px,
    borderTop: `1px solid ${ theme.colors.gray.grayLight.hex }`,
  },
  text: {
    fontSize: 12,
    textTransform: 'capitalize',
  },
  link: {
    color: 'inherit',
    '&:hover': {
      color: theme.colors.primary.indigoBlue.hex,
    },
  },
})

const Typefaces = ({ classes }) => (
  <div className={classes.root}>
    {Object.keys(theme.typography.typefaces).map((elem, idx) => (
      <div className={classes.container} key={`typeface${ idx }`}>
        <div className={classes.titleContainer}>
          <p className={classes.title} style={{ fontFamily: elem }}>
            Lorem Ipsum
          </p>
        </div>
        <div className={classes.buttonContainer}>
          <p className={classes.text}>{elem} Font Family</p>
          <a
            href={`https://fonts.google.com/specimen/${ elem.charAt(0).toUpperCase() +
              elem.slice(1).toLowerCase() }`}
            target="_blank"
            rel="noopener"
            aria-label="Launch"
            className={classes.link}
          >
            <OpenInNewIcon width="24" height="24" />
          </a>
        </div>
      </div>
    ))}
  </div>
)

Typefaces.propTypes = {
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(Typefaces)
