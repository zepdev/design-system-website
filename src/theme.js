import { createMuiTheme } from '@material-ui/core/styles'
import guidelines from '@zlab-de/zel-react/guidelines.json'

const theme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      light: guidelines.theme.indigo.secondary,
      main: guidelines.theme.indigo.primary,
      dark: guidelines.theme.indigo.primary,
    },
    error: { main: guidelines.color.support.alert },
  },
  status: {
    gray: '#eceeef',
  },
  logo: guidelines.logo,
  blue: guidelines.theme.blue,
  indigo: guidelines.theme.indigo,
  red: guidelines.theme.red,
  yellow: guidelines.theme.yellow,
  guidelines: guidelines,
  space: guidelines.spacing.component,
  color: guidelines.color,
})

export default theme
