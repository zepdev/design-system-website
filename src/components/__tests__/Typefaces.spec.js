import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'react-jss'
import theme from '../../data/theme'
import Typefaces from '../Typefaces'

describe('Typefaces', () => {
  it('renders correctly', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Typefaces />
      </ThemeProvider>
    )
    expect(component).toMatchSnapshot()
  })
})
