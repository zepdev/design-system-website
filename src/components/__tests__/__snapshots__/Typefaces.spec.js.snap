// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typefaces renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Typefaces-root-0-1-1"
      >
        <div
          class="Typefaces-container-0-1-2"
        >
          <div
            class="Typefaces-titleContainer-0-1-3"
          >
            <p
              class="Typefaces-title-0-1-4"
              style="font-family: roboto;"
            >
              Lorem Ipsum
            </p>
          </div>
          <div
            class="Typefaces-buttonContainer-0-1-5"
          >
            <p
              class="Typefaces-text-0-1-6"
            >
              roboto
               Font Family
            </p>
            <a
              aria-label="Launch"
              class="Typefaces-link-0-1-7"
              href="https://fonts.google.com/specimen/Roboto"
              rel="noopener"
              target="_blank"
            >
              <svg
                aria-labelledby="title"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title
                  id="title_launch"
                >
                  Launch Icon
                </title>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
        <div
          class="Typefaces-container-0-1-2"
        >
          <div
            class="Typefaces-titleContainer-0-1-3"
          >
            <p
              class="Typefaces-title-0-1-4"
              style="font-family: teko;"
            >
              Lorem Ipsum
            </p>
          </div>
          <div
            class="Typefaces-buttonContainer-0-1-5"
          >
            <p
              class="Typefaces-text-0-1-6"
            >
              teko
               Font Family
            </p>
            <a
              aria-label="Launch"
              class="Typefaces-link-0-1-7"
              href="https://fonts.google.com/specimen/Teko"
              rel="noopener"
              target="_blank"
            >
              <svg
                aria-labelledby="title"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title
                  id="title_launch"
                >
                  Launch Icon
                </title>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                  fill="currentColor"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Typefaces-root-0-1-1"
    >
      <div
        class="Typefaces-container-0-1-2"
      >
        <div
          class="Typefaces-titleContainer-0-1-3"
        >
          <p
            class="Typefaces-title-0-1-4"
            style="font-family: roboto;"
          >
            Lorem Ipsum
          </p>
        </div>
        <div
          class="Typefaces-buttonContainer-0-1-5"
        >
          <p
            class="Typefaces-text-0-1-6"
          >
            roboto
             Font Family
          </p>
          <a
            aria-label="Launch"
            class="Typefaces-link-0-1-7"
            href="https://fonts.google.com/specimen/Roboto"
            rel="noopener"
            target="_blank"
          >
            <svg
              aria-labelledby="title"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="title_launch"
              >
                Launch Icon
              </title>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
      <div
        class="Typefaces-container-0-1-2"
      >
        <div
          class="Typefaces-titleContainer-0-1-3"
        >
          <p
            class="Typefaces-title-0-1-4"
            style="font-family: teko;"
          >
            Lorem Ipsum
          </p>
        </div>
        <div
          class="Typefaces-buttonContainer-0-1-5"
        >
          <p
            class="Typefaces-text-0-1-6"
          >
            teko
             Font Family
          </p>
          <a
            aria-label="Launch"
            class="Typefaces-link-0-1-7"
            href="https://fonts.google.com/specimen/Teko"
            rel="noopener"
            target="_blank"
          >
            <svg
              aria-labelledby="title"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="title_launch"
              >
                Launch Icon
              </title>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
